mig/main.tf

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  distribution_policy_zones = {
+    default = ["${data.google_compute_zones.available.names}"]
+    user    = ["${var.distribution_policy_zones}"]
+  }
+}
+
+data "google_compute_zones" "available" {}
+
+resource "google_compute_region_instance_group_manager" "mig" {
+  provider           = "google-beta"
+  count              = "${var.mig_enabled ? 1 : 0}"
```

Comment:
`var.mig_enabled` is redundant since this module is specifically designed to manage a MIG.